### PR TITLE
Implemented UI for the Tab displaying infos on an activity when clicked

### PR DESCRIPTION
**Pull request by ecornamu**
The UI looks like this 
image
and is based on the Figma mockup for this tab.
- It is already tested for a line coverage of 80% (can be improved if needed)
- The current state of the app doesn't include an `address` attribute for `OutdoorActivity` nor a picture of the activity. Thus, both features are not displayed on the UI yet but can be done as soon as those attributes are a thing.
- Same for the `Community` tab, implemented as a `Spacer` here but can be added afterwards
- Since this is strictly UI, every button doesn't do anything yet.
- There is a helper function that should be put in a `ViewModel`, which I will do later but since it doesn't change functionality I instantiated the PR without this change but it will be done

### CODE REVIEW

File: app/src/main/res/drawable/archive.png
Comment (Andy130604) [inline app/src/main/res/drawable/archive.png:1]: This is a download button, not archive make sure to rename this

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:1]: We'll have to look into naming of these files, maybe have the screen called MoreInfoScreen.kt inside moreinfo package would make more sense. You could have a composable named ActivityDisplay or whatever.

**Review by Thimphou — CHANGES_REQUESTED** [review]
Hi Emile! 

Your UI work looks great, it is tested, and the Android tests look fine to me. 

Your code is commented, and maybe restructuring as Andrew said would increase our code usability. 

Now that the Color theme implementation is merged on main, I would like you not to hardcode the Colors anymore, but start coding it with the app's colors. 

Otherwise, great code!

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+fun StartButton() {
+  Row(
+      modifier = Modifier.fillMaxWidth().testTag("Start button"),
+      horizontalArrangement = Arrangement.Center) {
+        ElevatedButton(
+            onClick = {
+              /** TODO : Start Activity */
+            },
+            modifier = Modifier.width(305.dp).height(48.dp),
+            colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF0096CF))) {
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:61]: Just change the color used here!

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+@Composable
+fun ViewOnMapButton() {
+  Column(modifier = Modifier.padding(vertical = 25.dp), horizontalAlignment = Alignment.End) {
+    ElevatedButton(
+        onClick = {
+          /** TODO : Go to map */
+        },
+        contentPadding = PaddingValues(all = 3.dp),
+        modifier = Modifier.width(130.dp).height(40.dp),
+        colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF0096CF))) {
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:92]: Here as well.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+  }
+}
+
+@Composable
+fun RatingDisplay(rating: Double) {
+  Row(modifier = Modifier.padding(vertical = 30.dp)) {
+    Icon(
+        imageVector = Icons.Filled.Star,
+        contentDescription = "Rating Star",
+        tint = Color(0, 150, 207, 255),
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:120]: Here, you could either use the "yellow" hardcoded value of the star, or the blue/green of our app.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  }
+}
+
+@Composable
+fun ElevatedDifficultyDisplay(diff: String) {
+  ElevatedButton(
+      onClick = {},
+      contentPadding = PaddingValues(all = 3.dp),
+      modifier = Modifier.width(80.dp).height(24.dp),
+      colors = ButtonDefaults.buttonColors(containerColor = Color(0xCCFFE713))) {
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:154]: Here as well.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        Text(
+            diff,
+            style =
+                TextStyle(
+                    fontSize = 16.sp,
+                    lineHeight = 24.sp,
+                    fontWeight = FontWeight(500),
+                    color = Color(0xFF000000),
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:162]: Here for the dark mode.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+@Composable
+fun TopBarLogo(logoPainterId: Int, f: () -> Unit) {
+  IconButton(onClick = { f() }) {
+    Icon(
+        painter = painterResource(id = logoPainterId),
+        contentDescription = "Top Bar logo",
+        modifier = Modifier.width(26.dp).height(26.dp),
+        tint = Color(0, 150, 207, 255))
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:186]: Here as well to use the green or blue of Lasta.

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+  }
+}
+
+@Composable
+fun ElevatedActivityType(activity: OutdoorActivity) {
+  ElevatedButton(
+      onClick = {},
+      contentPadding = PaddingValues(all = 3.dp),
+      modifier = Modifier.padding(3.dp).width(64.dp).height(20.dp),
+      colors = ButtonDefaults.buttonColors(containerColor = Color(0, 150, 207, 128))) {
```
Comment (Thimphou) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/activities/ActivityScreen.kt:235]: And lastly, here!

**Review by Thimphou — COMMENTED** [review]
I added a few comments on your code.

**Review by Thimphou — APPROVED** [review]

**Comment by ecornamu** [comment]
Handled both requests ! With more details : 
- Colors of the UI are now using either the lasta theme default ones or the ones defined in `Color.kt`
- Image is now renamed `download_button` (previously `archive`, pickiest request I've had lol)
- The `MoreInfoScreen` is now connected to both the `FavoritesScreen` and the `DiscoveryScreen` when you click on an activity. For the moment, a fake activity is hard-coded and displayed but it will be replaced by the real requested activity when we have a view model

**Comment by Thimphou** [comment]
Life is now better than ever! Thank you for your changes :)

**Review by Andy130604 — APPROVED** [review]
These are some nice changes, not so sure about your navigation implementation, I hope you'll look into it for the next sprint. Otherwise really nice design

**Comment by ecornamu** [comment]
Last commits were because @Andy130604 and I were trying to fix a `test` not passing, whose solution ended up being to clear the cache of the CI server

**Pull request merged**